### ISSUE TITLE
Allow user to persist --console setting in gradle.properties

### ISSUE THREAD

**Issue by oehme** [gradle/gradle#2744]

**Comment by adammurdoch** [comment]
When implementing, please do this: https://github.com/gradle/gradle/issues/1946 rather than adding even more ad hoc code that behaves slightly differently to everything else.

**Comment by pranayhasan** [comment]
This is a feature necessary for users like us to migrate from **gradle2.x** to **gradle 4.x**
Most of the I/O features are being affected as the gradle daemon gets along the console output for which our users have to explicitly specify this **--console plain**. 
It would be great if users have an option to add a property to _gradle.properties_ once and for all.

**Comment by oehme** [comment]
@pranayhasan Could you please elaborate what you mean? `--console plain` only affects the output that's printed on the console. It is not related to the daemon.

**Comment by pranayhasan** [comment]
@oehme We have are using Gradle 2.x and are looking to migrate to 4.x. Some custom gradle tasks that we created like **azkabanUpload** runs with the gradle build and takes input from users by printing some questionnaire. The gradle's build output gets along with the output of the questionnaire for which we had to do some hacks like [this](https://github.com/linkedin/linkedin-gradle-plugin-for-apache-hadoop/blob/master/hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanHelper.groovy#L221) where **sleep()** is being called so that the output buffer is flushed to the console.

The migration from 2.x to 4.x changes the way the console output behaves and hence we want to have a gradle property by which **--console plain** can be enforced on all the projects we have.

Yes, its true that having an API to ask for input solves all our hacky solutions.
